packages/app/src/vis-packs/core/complex/MappedComplexHeatmapVis.tsx

```diff
@@ -58,26 +58,20 @@ function MappedComplexHeatmapVis(props: Props) {
     invertColorMap,
   } = config;
 
+  const { phase, amplitude } = usePhaseAmplitude(value);
   const numAxisArrays = useToNumArrays(axisValues);
 
-  const [slicedDims, slicedMapping] = useSlicedDimsAndMapping(dims, dimMapping);
-  const mappedArray = useMappedArray(value, slicedDims, slicedMapping);
+  const slicedArgs = useSlicedDimsAndMapping(dims, dimMapping);
+  const phaseArray = useMappedArray(phase, ...slicedArgs);
+  const amplitudeArray = useMappedArray(amplitude, ...slicedArgs);
 
-  const { phaseValues, phaseBounds, amplitudeValues, amplitudeBounds } =
-    useMemo(() => getPhaseAmplitudeValues(mappedArray.data), [mappedArray]);
```

Comment:
Note that we now compute the phase/amplitude domains in separate loops. But this happens on first render and never again, so it's acceptable.

It's definitely better than to recompute the phase/amplitude values and domains every time the user remaps the dimensions (since previously, `useMappedArray` was called before `getPhaseAmplitudeValues`).